quickstarts/08-data-agent-mcp-chainlit/README.md

```diff
@@ -0,0 +1,163 @@
+# A conversational agent over a Postgres database using MCP
+
+This quickstart demonstrates how to build a fully functional, enterprise-ready agent that allows users to ask their database any question in natural text and get both the results and a highly structured analysis of complex questions. This quickstart also shows the usage of MCP in Dapr Agents to connect to the database and provides a fully functional ChatGPT-like chat interface using Chainlit.
+
+## Key Benefits
+
+- **Conversational Knowledge Base**: Users can talk to their database in natural language, ask complex questions and perform advanced analysis over data
+- **Conversational Memory**: The agent maintains context across interactions in the user's [database of choice](https://docs.dapr.io/reference/components-reference/supported-state-stores/)
+- **UI Interface**: Use an out-of-the-box, LLM-ready chat interface using [Chainlit](https://github.com/Chainlit/chainlit)
+- **Boilerplate-Free DB Layer**: MCP allows the Dapr Agent to connect to the database without requiring users to write Postgres-specific code
+
+## Prerequisites
+
+- Python 3.10 (recommended)
+- pip package manager
+- OpenAI API key (for the OpenAI example)
+- [Dapr CLI installed](https://docs.dapr.io/getting-started/install-dapr-cli/)
+
+## Environment Setup
+
+```bash
+# Create a virtual environment
+python3.10 -m venv .venv
+
+# Activate the virtual environment 
+# On Windows:
+.venv\Scripts\activate
+# On macOS/Linux:
+source .venv/bin/activate
+
+# Install dependencies
+pip install -r requirements.txt
+
+# Initialize Dapr
+dapr init
+```
+
+## LLM Configuration
+
+For this example, we'll be using the OpenAI client that is used by default. To target different LLMs, see [this example](../02_llm_call_dapr/README.md).
+
+Create a `.env` file in the project root:
+
+```env
+OPENAI_API_KEY=your_api_key_here
+```
+
+Replace `your_api_key_here` with your actual OpenAI API key.
+
+## Postgres Configuration
+
+### Connect to an existing database
+
+Create an `.env` file in the root directory of this quickstart and insert your database configuration:
+
+```bash
+DB_HOST=<HOST>
+DB_PORT=<PORT>
+DB_NAME=<DATABASE-NAME>
+DB_USER=<USER>
+DB_PASSWORD=<PASSWORD>
+```
+
+### Create a new sample database
+
+First, install Postgres on your machine.
+
+#### Option 1: Using Docker
+
+Create the following direcotry and copy the sql files there:
+
+```bash
+mkdir docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d
```

Comment:
this needs to be `mkdir docker-entrypoint-initdb.d`. why dont just have the folder already as part of the quickstart folder 😅 with the SQL files already there.

Reply:
ahh I guess we cant put the SQL files in the folder because they can be used by the option 2 to create / set up the postgres DB. However, the creation of the folder at least needs to be `mkdir docker-entrypoint-initdb.d` and not `mkdir docker-entrypoint-initdb.d:/docker-entrypoint-initdb.d`

Reply:
Why not do one of these:

1. (favorite) Change option 2 to point to the files in the `docker-entrypoint-initdb.d/` folder
2. Create the folder and add symlinks* to the files in that folder

\* While symlinks won't work when browsing the directories on GitHub it should work just finely when pulling down the codebase.

Reply:
Yeah option 1 makes sense. I just did not know if @yaron2 wanted to avoid the `docker` context when talking about non-docker deployment 😅

Reply:
> Yeah option 1 makes sense. I just did not know if @yaron2 wanted to avoid the docker context when talking about non-docker deployment 😅

Yeah that's my beef with that option. I'm gonna merge it now so we can demo in the community call and then get back to this and try and find another solution :)